### PR TITLE
Some Smarter Fridges

### PR DESCRIPTION
**Pull request by Youtubeboy139**
## About The Pull Request

Long before, I had created the early rendition of this sprite with the intent of implementing it however that never happened so I said 'fuck it' and decided to implement it now. I refined the sprite (looks a lot better than my first draft which was disproportionate as hell) and imported it into the code tweaking said code so that the on/off states worked properly (it's a glass cover, you should see the materials still inside even when the machine is unpowered - just darker). Took a few hours but I am ultimately happy with how this turned out. Oh, also added the missing `update_icon()` code line to the `smartfridge` code so when you actually empty the smart fridge it, you know, looks empty. Don't know if that was an issue before, but I encountered said issue while testing.

_I am aware that the current storage is mounted inside of a wall and this first sprite will look funky at first but know that the next thing (related to this, I might focus on another project) I am working on is a fulltile fridge similar to what TG currently has (a full 32x32 icon) so that even the current fridge looks better when mounted in a wall._

EDIT:
Decided that I may as well just bite the bullet and make the 'wall mounted' sheet storage so I created two extra sheet storages, a blank Industrial Sheet Storage meant for the Engineering department and a striped Mining Sheet Storage for the Mining department. I want to make it clear that **_despite the name and appearance they are not bigger than the original sheet storage_** but exist as alternate sprites for the original sheet storage and are meant to be wall mounted. For the time being these new Industrial Sheet Storages only exist in the code, they are not mapped in as I do not want to deal with mapping conflicts until they're merged.

EDIT 2:
Whoopsie I also did the regular Smart Fridges as well. They work similar to above and all have custom overlays that give players a visual representation of what is being stored inside the fridges. I also threw in a Chef Smart Fridge because for whatever reason there was no fridge that could store food items - just seeds and produce. Also, while I was in there I changed around the unique Showcase fridge - tweaked the sprite, made the hologram flicker and more transparent.

## How This Contributes To The VOREStation Roleplay Experience

Adds a custom sprite for the unique Smart Sheet Storage because it makes better sense sheets of steel and other materials would be kept in a more industrial looking machine and not a commercial fridge. Also fixed an non-issue issue in the code while I was at it so that might help better the experience as well.

EDIT:
Adds in a custom Smart Fridge sprite that isn't just the Eris sprite ported over and thus gives maintainers a more organic feel to the machines (no longer will the Smart Sheet storage or any Smart Fridge for that matter have to be haphazardly merged into a wall Turf just to make the scene look completed)

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

Example of Wall Mounted Smart Fridges (on) in a Test Session

Example of Wall Mounted Smart Fridges (off) in a Test Session

Examples of all Smart Fridges (on) in a Test Session

Examples of all Smart Fridges (off) in a Test Session

Screencap of the new `vending_smartfridge.dmi`
Screenshot 2023-05-25 214122

</details>

## Changelog

:cl:
add: Added new overlay mechanic for all the Smart Fridges
add: Added new Industrial Sized Sheet Storages for wall mounting
add: Added Chef Smart Fridge
qol: Organized files in the Smart Fridge folder so items are easier to locate
qol: Moved all Smart Fridges to their own `.dmi` to prevent excessive bloat on the existing `vending.dmi` file
image: Added new base Smart Sheet Storage sprite
image: Added new Mining and Industrial Sheet Storage sprites
image: Added new Smart Fridge sprites
image: Added new Smart Showcase sprites
fix: Made it to where the `smartfridge` actually updates its icon when you empty it
del: Removed unsecure Viro Smart Fridge
del: Removed unsecure Chem Smart Fridge
/:cl:

### CODE REVIEW

**Comment by KillianKirilenko** [comment]
Whilst I'm not really sold on the little standalone smartsheet machine, all the other full-size ones look good! Especially with the filled states.

**Comment by Youtubeboy139** [comment]
> Whilst I'm not really sold on the little standalone smartsheet machine, all the other full-size ones look good! Especially with the filled states.

Item Removed

**Comment by Heroman3003** [comment]
The sprites seem good to me, although I would like to see the 'unresistricted' chem/virus storage not be removed. They are unused, but 'unlocked' subtypes are simple and I don't see issues with them existing for potential mapping purposes. Beyond that this seems fine.

**Comment by Youtubeboy139** [comment]
> The sprites seem good to me, although I would like to see the 'unresistricted' chem/virus storage not be removed. They are unused, but 'unlocked' subtypes are simple and I don't see issues with them existing for potential mapping purposes. Beyond that this seems fine.

Items Re-added

**Comment by TankTheBirb** [comment]
Good sprites, these looks excellent. Especially compared to using the same sprite for everything like before. Unique sprites are very helpful to players and are a good thing to have regardless.

**Comment by Youtubeboy139** [comment]
I don't know what I did, but I apparently broke something and I can't find what

File: code/modules/food/kitchen/smartfridge/medical.dm [hunk 1]
```diff
@@ -1,21 +1,46 @@
+/*
+ * Medical
+ */
+ /obj/machinery/smartfridge/medbay
```
Comment (KillianKirilenko) [inline code/modules/food/kitchen/smartfridge/medical.dm:4]: erroneous space at the start of this line

Comment (Youtubeboy139) [reply code/modules/food/kitchen/smartfridge/medical.dm]: Fixed that but the check is still angry with me

**Comment by Youtubeboy139** [comment]
Found it, Fixed

**Pull request merged**